### PR TITLE
Integrate Towncrier configuration

### PR DESCRIPTION
**Pull request by webknjaz**

### CODE REVIEW

**Comment by webknjaz** [comment]
@abhinavsingh this adds the basic infrastructure for the changelog management. To use it, currently, you have to manually run `tox -e make-changelog -- <your desired version>` and then commit and push the changes (which will `git rm` the fragments and render them in `CHANGELOG.md`).
You can also preview a draft (printed out in the console output) if you skip the arguments, to do this, just run `tox -e make-changelog`.

File: .github/chronographer.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (webknjaz) [inline .github/chronographer.yml:1]: Add https://github.com/apps/psf-chronographer to make use of this config. It's a simple check for whether the PR authors didn't forget to add a change note. The config is currently set up to ignore PRs from known bots.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<!-- markdownlint-disable no-duplicate-heading -->
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+<!-- towncrier release notes start -->
```
Comment (webknjaz) [inline CHANGELOG.md:9]: @abhinavsingh do not remove this line. This is how Towncrier will know where to inject the generated changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<!-- markdownlint-disable no-duplicate-heading -->
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
```
Comment (webknjaz) [inline CHANGELOG.md:7]: The block of text above the marker is editable and static. But do not touch the contents below the marker. Maybe except for one-time consistency adjustment.

File: docs/changelog-fragments.d/823.misc.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Added changelog fragment management infrastructure using [Towncrier]
```
Comment (webknjaz) [inline docs/changelog-fragments.d/823.misc.md:1]: @abhinavsingh this is an example change note. Markdown is optional but should be valid if present. Same for the signature.

File: docs/changelog-fragments.d/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# Adding change notes with your PRs
```
Comment (webknjaz) [inline docs/changelog-fragments.d/README.md:1]: This document shows up on GitHub when somebody navigates to the change notes directory: https://github.com/webknjaz/proxy.py/blob/maintenance/towncrier/docs/changelog-fragments.d/.
You can point the PR authors here.

Comment (webknjaz) [reply docs/changelog-fragments.d/README.md:1]: It's also included in the Sphinx docs.

File: pyproject.toml [hunk 6]
```diff
@@ -11,3 +11,50 @@ build-backend = "setuptools.build_meta"
...
+
+  [[tool.towncrier.type]]
+    directory = "misc"
+    name = "Miscellaneous"
+    showcontent = true
+
+  [[tool.towncrier.type]]
+    directory = "internal"
+    name = "Contributor-facing changes"
+    showcontent = true
```
Comment (webknjaz) [inline pyproject.toml:60]: These are the categories for possible changelog subsections. The fragment files should be named `<issue/pr number>.<category from this list>.rst` for Towncrier to properly work with it. If you want a note under several categories, make a symlink with an additional name.

**Comment by webknjaz** [comment]
@abhinavsingh I think I've fixed all the spelling offenses and this should be good to go now.

File: .github/chronographer.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+enforce_name:
+  suffix: .md
+exclude:
+  bots:
+  - dependabot-preview
+  - dependabot
+  - patchback
+  humans:
+  - pyup-bot
```
Comment (abhinavsingh) [inline .github/chronographer.yml:10]: jFYI, I have removed all `pyup` integrations.

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+<!-- towncrier release notes start -->
+
+
+## v2.4.0
+
+- No longer support `Python 3.6` due to `asyncio.run` usage in the core.
```
Comment (abhinavsingh) [inline CHANGELOG.md:14]: Wow I forgot to remove this line :(

Comment (webknjaz) [reply CHANGELOG.md]: This is one of the reasons to have a proper release notes management :)

**Pull request merged**